src/sdi-refresh-monitor.c

```diff
@@ -439,8 +444,15 @@ static void manage_change_update(SnapdClient *source, GAsyncResult *res,
   if (change == NULL)
     return;
 
-  gboolean done = g_str_equal(snapd_change_get_status(change), "Done");
-  gboolean cancelled = cancelled_change_status(snapd_change_get_status(change));
+  const gchar *change_status = snapd_change_get_status(change);
+
+  gboolean done = g_str_equal(change_status, "Done");
+  gboolean cancelled = cancelled_change_status(change_status);
+  gboolean valid_do = valid_working_change_status(change_status);
+  if (!(valid_do || cancelled)) {
+    g_print("Unknown state %s\n", change_status);
```

Comment:
g_warning or g_debug instead?

Reply:
Also the text could probably be more specific.

Reply:
Good point.